api/helpers/imageProcessor.js

```diff
@@ -0,0 +1,409 @@
+const crypto = require('crypto');
+const { createBlockBlobFromText } = require('./blob');
+
+// Blob container name
+const BLOB_CONTAINER_NAME = process.env.BLOB_CONTAINER_NAME || 'cblob';
+
+// Configuration
+const CONFIG = {
+  BATCH_SIZE: 5,
+  MAX_RETRIES: 3,
+  ENABLE_CACHE: true
+};
+
+// Error types for classification
+const ErrorTypes = {
+  INVALID_BASE64: 'Invalid base64 format',
+  AZURE_UPLOAD_FAILED: 'Azure blob upload failed',
+  AZURE_RATE_LIMIT: 'Azure rate limit exceeded',
+  NETWORK_TIMEOUT: 'Network timeout',
+  INSUFFICIENT_STORAGE: 'Storage quota exceeded',
+  INVALID_IMAGE_FORMAT: 'Unsupported image format'
+};
+
+// Global image cache - maps content hash to blob URL
+const imageCache = new Map();
```

Comment:
Please remove this `imageCache` map. It can produce a memory leak because it can grow indefinitely. It never gets cleaned.

Also I think that not to many images are duplicated, so we will be good if each blob is processed